### PR TITLE
convert proxy to optional dep; allow direct connection to bitcoind

### LINKED ISSUE #17: Make C-lightning capable of using bitcoind directly, without proxy
This is so bitcoind will be enforced as a dependency of c-lightning only if bitcoind is in full archival mode

### PR DESCRIPTION
**Pull request by chrisguida**
Fixes #17

### CODE REVIEW

File: check-synced.sh [hunk 1]
```diff
@@ -2,11 +2,28 @@
 
 set -e
 
-b_username=$(yq e '.bitcoind.user' /root/.lightning/start9/config.yaml)
-b_password=$(yq e '.bitcoind.password' /root/.lightning/start9/config.yaml)
+# if ! [ -s /root/.lightning/config ]; then
```
Comment (elvece) [inline check-synced.sh:5]: delete or uncomment

File: manifest.yaml [hunk 2]
```diff
@@ -194,6 +194,45 @@ dependencies:
         mounts:
           compat: /mnt/assets
           main: /root/.lightning
+  bitcoind:
+    version: ">=0.21.1.2 <23.0.0"
+    requirement:
+      type: "opt-out"
+      how: Can alternatively configure an internal Bitcoin Proxy, or an external bitcoin node.
```
Comment (elvece) [inline manifest.yaml:201]:
```suggestion
      how: Can alternatively configure the internal Bitcoin Proxy service, or an external bitcoin node.
```

**Comment by elvece** [comment]
In order for proxy/bitcoin to display in the UI as a required dependency *if* selected in the config, you will need to add the [dependencies rules file](https://github.com/Start9Labs/btcpayserver-wrapper/blob/b8096529b0ed3230e881590cf73581fb17b3570d/assets/compat/dependencies.yaml). Then, call this file in the config.set action, [like so](https://github.com/Start9Labs/btcpayserver-wrapper/blob/b8096529b0ed3230e881590cf73581fb17b3570d/manifest.yaml#L90).

**Review by elvece — CHANGES_REQUESTED** [review]
In order for proxy/bitcoin to display in the UI as a required dependency *if* selected in the config, you will need to add the [dependencies rules file](https://github.com/Start9Labs/btcpayserver-wrapper/blob/b8096529b0ed3230e881590cf73581fb17b3570d/assets/compat/dependencies.yaml#L9-L11). Then, call this file in the config.set action, [like so](https://github.com/Start9Labs/btcpayserver-wrapper/blob/b8096529b0ed3230e881590cf73581fb17b3570d/manifest.yaml#L90).

**Comment by chrisguida** [comment]
@elvece my understanding is that dependencies.yaml is not necessary if there's a pointer to bitcoind

**Comment by ProofOfKeags** [comment]
No additional comments beyond what @elvece pointed out.

File: manifest.yaml (hunk 2 above)
Comment (chrisguida) [reply manifest.yaml]: Done

File: check-synced.sh (hunk 1 above)
Comment (chrisguida) [reply check-synced.sh]: Done

**Pull request merged**